src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -313,8 +307,7 @@ void goto_symext::symex_printf(
   statet &state,
   const exprt &rhs)
 {
-  if(rhs.operands().empty())
-    throw "printf expected to have at least one operand";
```

Comment:
I have made the judgement call to drop what I consider to be redundant error messages, in the cases where the `PRECONDITION` assertion itself (within the context) seems self evident.